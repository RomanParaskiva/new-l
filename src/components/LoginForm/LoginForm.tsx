import React, { ChangeEvent, useState } from "react";
import { useAuth } from "../../hooks/auth.hook";

export const LoginForm = () => {
  const [name, setName] = useState("");
  const { login } = useAuth();

  const handleInput = (e: ChangeEvent<HTMLInputElement>) => {
    setName(() => e.target.value);
  };

  const handleStart = () => {
    if (name.length > 3) {
      login(name);
    }
  };

  return (
    <div className="flex w-100 py-20 justify-center items-center">
      <div className="bg-gray-200 rounded p-3">
        <form
          onSubmit={(e) => e.preventDefault()}
          className="flex flex-col gap-5"
        >
          <label className="text-center font-semibold text-lg">Ваше имя</label>
          <input
            data-testid="userName"
            className="outline-none text-black bg-transparentborder-gray-600 rounded-md border p-1"
            type="text"
            value={name}
            name="userName"
            aria-label="userName"
            onChange={handleInput}
          />

          <button
            className="p-3 bg-green-400 rounded-md text-white uppercase "
            onClick={handleStart}
          >
            start
          </button>
        </form>
      </div>
    </div>
  );
};
