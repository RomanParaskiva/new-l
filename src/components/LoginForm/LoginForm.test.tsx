import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import { LoginForm } from "./LoginForm";

describe("test LoginForm", () => {
    test("renders LoginForm component", async () => {
        render(<LoginForm />);

        expect(await screen.findByText(/start/)).not.toBeNull();
    });

    test("LoginForm snapshot", () => {
        const { container } = render(<LoginForm />);

        expect(container).toMatchSnapshot();
    })

    const setup = () => {
        const utils = render(<LoginForm />)
        const input = utils.getByLabelText('userName')
        return {
          input,
          ...utils,
        }
      }

    test("test input change", async () => {
        render(<LoginForm />);

        const input = screen.getByTestId("userName");

        await fireEvent.input(input, { target: { value: "Roman" }});

        await waitFor(() => expect( input ).toHaveDisplayValue("Roman"))
        
    })
});