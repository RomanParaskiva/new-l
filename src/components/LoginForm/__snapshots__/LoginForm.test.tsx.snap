// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LoginForm LoginForm snapshot 1`] = `
<div>
  <div
    class="flex w-100 py-20 justify-center items-center"
  >
    <div
      class="bg-gray-200 rounded p-3"
    >
      <form
        class="flex flex-col gap-5"
      >
        <label
          class="text-center font-semibold text-lg"
        >
          Ваше имя
        </label>
        <input
          aria-label="userName"
          class="outline-none text-black bg-transparentborder-gray-600 rounded-md border p-1"
          data-testid="userName"
          name="userName"
          type="text"
          value=""
        />
        <button
          class="p-3 bg-green-400 rounded-md text-white uppercase "
        >
          start
        </button>
      </form>
    </div>
  </div>
</div>
`;
